src/main/java/com/marklogic/spark/Options.java

```diff
@@ -21,6 +21,8 @@ public abstract class Options {
     public static final String READ_XQUERY = "spark.marklogic.read.xquery";
     public static final String READ_XQUERY_FILE = "spark.marklogic.read.xqueryFile";
     public static final String READ_VARS_PREFIX = "spark.marklogic.read.vars.";
+    public static final String READ_PARAMS_PREFIX = "spark.marklogic.read.params.";
+    public static final String READ_COLUMN_PARAMS_PREFIX = "spark.marklogic.read.columnParams.";
```

Comment:
As for the name - I agree adding optic makes sense.  I mostly just followed the pattern of what I saw already there.  The `params.{param name}={default value}` pattern is how it currently works.  I used the name columnParams to hopefully indicate that it would create a column that it uses to get the value (and to differentiate it from the hard-coded param option you commented on above), but I agree that the name isn't very communicative.  I think `spark.marklogic.read.optic.params.{name}` is fine, especially if we aren't keeping the hardcoded option.  I could also see `filterParams` or `pushdownParams` as being a better name than `columnParams`  People will have to read the documentation to figure out how to use it regardless.